profiles/dev/user.clj

```diff
@@ -36,3 +36,11 @@
          [:message-page :cache-time]
          (constantly new-cache-time))
   true)
+
+(comment
+  (go)
+  (reset)
+  (reset-all)
+  (use 'clojurians-log.repl)
+  (load-demo-data! "../clojurians-log-demo-data")
+  #_:end)
```

Comment:
Useful snippets for repl-driven development in Sapcemacs. Question: is there a better place to put this? Or do you bind those to some keystrokes in your IDE? Or in general: which keystrokes do you bind to which commands for repl-driven development?

Reply:
It's fine to have it in a comment here. Alternatively you can put it in a file in the `repl/` directory.

I also use Spacemacs, with standard bindings like `, e e` to eval. I also have some leader key bindings myself.

```
(defun plexus-clojure-extras/cider-pprint-register (register)
  (interactive (list (register-read-with-preview "Eval register: ")))
  (cider--pprint-eval-form (get-register register)))

(dolist (m '(clojure-mode
               clojurec-mode
               clojurescript-mode
               clojurex-mode
               cider-repl-mode
               cider-clojure-interaction-mode))

    (spacemacs/set-leader-keys-for-major-mode m
      "ep" 'cider-pprint-eval-last-sexp
      "eP" 'cider-pprint-eval-last-sexp-to-comment
      "en" 'cider-eval-ns-form
      "er" 'cider-eval-last-sexp-and-replace
      ","  'plexus-clojure-extras/cider-pprint-register
      "lp" 'sesman-link-with-project
      "lb" 'sesman-link-with-buffer
      "lb" 'sesman-link-with-directory
      "ll" 'sesman-link-with-least-specific
      "ss" (if (eq m 'cider-repl-mode)
               'cider-switch-to-last-clojure-buffer
             'cider-switch-to-repl-buffer)
      "'"  'cider-jack-in-clj
      "\"" 'cider-jack-in-cljs
      "\&" 'cider-jack-in-clj&cljs
      "sq" 'cider-quit
      ))
```

I use `, e p` a lot. `, e P` will insert the result in the buffer in a comment, or you can use `SPC u , e e` to insert directly.

The bottom ones are to undo the recent keybinding changes in the spacemacs clojure layer.

I also have a special `, , <register>` binding, so I can put snippets in a register and eval them from anywhere. I don't use it a lot, but sometimes it's very handy.